psa_car_controller/psacc/utils/utils.py

```diff
@@ -14,7 +14,7 @@ def get_temp(latitude: str, longitude: str, api_key: str) -> float:
                                                "exclude": "minutely,hourly,daily,alerts",
                                                "appid": api_key,
                                                "units": "metric"})
-            temp = weather_rep.json()["current"]["temp"]
+            temp = weather_rep.json()["main"]["temp"]
```

Comment:
Are you sure this is correct? The official [documentation ](https://openweathermap.org/api/one-call-api)it should be under `["current"]["temp"]`.

However, if I look at [this ](https://openweathermap.org/current)documentation, your change would be correct. Then we'd also need to change the url to `https://api.openweathermap.org/data/2.5/weather`. I would argue for this change, because the one call API is not free, and this one is.

Reply:
I tested it with the https://api.openweathermap.org/data/2.5/onecall api not the https://api.openweathermap.org/data/2.5/weather api. Perhaps openweather is somehow choosing to redirect to the free api since I don't have a paid plan?

Reply:
Mhm I see that the response in my test did return `X-Cache-Key: /data/2.5/weather?exclude=minutely,hourly,daily,alerts&lat=xx&lon=yy&units=metric`, perhaps I called the wrong api in my test.

Reply:
I think so because the api work well for me.

Reply:
Just an idea: If API call fails for **/onecall**, the next step would be to make an API call for **/weather** before going to error handling.
This way the free and the paid plan would give temperature results.